### PR TITLE
Various unit test/debugging fixes

### PR DESCRIPTION
**Pull request by martinpitt**
This is some stuff that fell  out from my investigations of PR #12580

### CODE REVIEW

File: pkg/playground/speed.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ var timer = null;
 var start = null;
 var total = 0;
 var proc = null;
+var close_problem;
```
Comment (marusak) [inline pkg/playground/speed.js:10]: `var`? :)

File: test/common/tap-cdp [hunk 2]
```diff
@@ -66,13 +67,18 @@ if not cdp.get_browser_path():
     print('1..0 # skip web browser not found')
     sys.exit(0)
 
+# pass the address through a separate fd, so that we can see g_debug() messages (which go to stdout)
+(addr_r, addr_w) = os.pipe()
```
Comment (marusak) [inline test/common/tap-cdp:71]: shouldn't you close those at the end?

File: test/common/tap-cdp [hunk 3]
```diff
@@ -66,13 +67,18 @@ if not cdp.get_browser_path():
...
+env["TEST_SERVER_ADDRESS_FD"] = str(addr_w)
+
 server = subprocess.Popen(opts.server,
-                          stdout=subprocess.PIPE,
-                          stdin=subprocess.PIPE,
-                          universal_newlines=True,
-                          close_fds=True)
+                          stdin=subprocess.DEVNULL,
+                          pass_fds=(addr_w,),
+                          close_fds=True,
```
Comment (marusak) [inline test/common/tap-cdp:78]: > Providing any pass_fds forces close_fds to be True

From here: https://docs.python.org/3/library/subprocess.html

And that also answers my above question :) (but the read one is still open, isn't it?)

File: pkg/playground/speed.js (hunk 1 above)
Comment (martinpitt) [reply pkg/playground/speed.js:10]: Let's keep the current file's code style. When I (or anyone else) is bored, we can convert it wholesale to some nice ES6 :grin:

File: test/common/tap-cdp (hunk 2 above)
Comment (martinpitt) [reply test/common/tap-cdp:71]: Indeed I should immediately close the write end after passing it to the subprocess. Fixed, thanks for spotting!

File: test/common/tap-cdp (hunk 3 above)
Comment (martinpitt) [reply test/common/tap-cdp:78]: right, I forgot to close that that after the read(). It pretty well doesn't matter, but for clarity/cleanliness I added a close for it as well. The subprocess only gets the write end, teh read end is closed automatically due to `closes_fds` indeed.

**Pull request merged**